godot-ffi/src/godot_ffi.rs

```diff
@@ -487,3 +487,67 @@ mod scalars {
         }
     }
 }
+
+unsafe impl<T> GodotFfi for Option<T>
```

Comment:
It compiles and Godot does not immediately crash when interacting with my dynlib. But beyond that I have no idea if this is even remotely correct.

Reply:
you're absolutely gonna have to write tests for this. Ideally tests a representative sample of all the variant types for this. Some ints, strings, floats, arrays, dictionaries, packed arrays, vector and matrix types, and of course Gd, but Gd is nullable anyway so it makes sense that it would work for Gd.

And dont forget to check *both* pointer and varcalls for that.

Reply:
Thanks for the feedback so far. Any pointers for how I should get started writing the tests?

Reply:
have a look through itest, and the build.rs script there for how to automatically generate some tests for passing stuff between godot and rust. you might need to do it manually though, there are some manual ffi tests in itest.

Reply:
I have now added ffi tests to `itest` that return and accept `Option<Gd<Node>>` and `Option<Gd<RefCounted>>`. Please let me know if there are test cases that should be covered.